src/st/no4/unk_55678.c

```diff
@@ -42,11 +40,748 @@ INCLUDE_ASM("st/no4/nonmatchings/unk_55678", func_us_801D8FE0);
+    u8 u0, u1;
+    s16 var_s4;
+    s16 var_s0_3;
+    s32 primIndex;
+    Tilemap* tilemap;
+    s16* point;
+    SVECTOR* points[4];
+    Point16* sxy[4];
+    u8* iptr;
+    CVECTOR* primaryColor;
+    CVECTOR* color;
+    VECTOR tempVector;
+    MATRIX* mat;
+    long* p;
+    long* flag;
+    Entity* tempEntity;
+    Primitive* prim;
+    Primitive* prim2;
+    s16 tempVar;
+    s16 i, j;
+
+    tilemap = &g_Tilemap;
+    self->ext.et_801D96FC.unk9E++;
+    if (self->step && !(self->flags & FLAG_DEAD)) {
+        self->ext.et_801D96FC.unk7C++;
+        if (self->ext.et_801D96FC.unk7C & 0x40) {
+            self->unk6C = (u8)(self->ext.et_801D96FC.unk7C & 0x3F) + 0x40;
+        } else {
+            self->unk6C = 0x80 - (u8)(self->ext.et_801D96FC.unk7C & 0x3F);
+        }
+    }
+    tempEntity = &PLAYER;
+    if ((self->flags & FLAG_DEAD) && (self->step != 4)) {
+        SetStep(4);
+    }
+    if (self->ext.et_801D96FC.unk82) {
+        prim = &g_PrimBuf[self->primIndex];
+        j = 0;
+        if (self->ext.et_801D96FC.unk82 != 2) {
+            if ((self->ext.et_801D96FC.unkA4 & 0x1F) == 0) {
+                PlaySfxPositional(SFX_GLASS_SHARDS);
+            }
+            self->ext.et_801D96FC.unkA4++;
+        }
+        while (prim != NULL) {
+            if (!prim->v2) {
+                if (!prim->p3) {
+                    if (self->facingLeft) {
+                        LOW(prim->u0) += D_us_80182840[prim->u2];
+                    } else {
+                        LOW(prim->u0) -= D_us_80182840[prim->u2];
+                    }
+                    LOW(prim->u1) +=
+                        D_us_80182860[prim->u2] - (prim->u3++ << 10);
+                    func_us_801D9560(prim);
+                    PRED(prim)--;
+                    PGRN(prim)--;
+                    PBLU(prim) -= 3;
+                    if (prim->p2) {
+                        tempEntity = &g_Entities[self->ext.et_801D96FC
+                                                     .unkA8[prim->p2 - 1]];
+                        if (tempEntity->hitFlags) {
+                            for (i = 0; i < 11; i++) {
+                                tempEntity->unk6D[i] = 0;
+                            }
+                            prim->drawMode = DRAW_HIDE;
+                            prim->v2 = 0x40 - prim->u3;
+                            prim = prim->next;
+                            continue;
+                        }
+                    }
+                    prim->p2 = 0;
+                    if (prim->p1 == (self->ext.et_801D96FC.unk9E & 3) &&
+                        prim->u3 < 0x30) {
+                        tempEntity =
+                            &g_Entities[self->ext.et_801D96FC.unkA8[j]];
+                        tempEntity->posX.i.hi =
+                            LOH(prim->clut) - tilemap->scrollX.i.hi;
+                        tempEntity->posY.i.hi =
+                            LOH(prim->tpage) - tilemap->scrollY.i.hi;
```

Comment:
Not a comment on PR quality, but these two seem rather suspicious to me... perhaps this is actually a Primitive variant and this is not actually using `clut` / `tpage`

Reply:
> Not a comment on PR quality, but these two seem rather suspicious to me... perhaps this is actually a Primitive variant and this is not actually using `clut` / `tpage`

`Primitive` sometimes seems to be used as a blob of data. This primitive is a `PRIM_G4`, an untextured coloured quad that does not make use of `clut`, `tpage`, `u0`, `v0`, and similar. @gamezter it's probably worth copy&paste the existing `Primitive` on top of this function and change name+type for these.